src/com/google/javascript/jscomp/deps/SortedDependencies.java

```diff
@@ -71,8 +71,12 @@
   }
 
   public static class MissingProvideException extends Exception {
-    MissingProvideException(String provide) {
+    public MissingProvideException(String provide) {
       super(provide);
     }
+
+    public MissingProvideException(String provide, Exception e) {
+      super(provide, e);
```

Comment:
It's caught in [Compiler.java](https://github.com/google/closure-compiler/blob/master/src/com/google/javascript/jscomp/Compiler.java#L1338) and translated to a user friendly message.